compiler/rustc_codegen_llvm/src/consts.rs

```diff
@@ -418,7 +414,7 @@ impl<'ll> StaticMethods for CodegenCx<'ll, '_> {
 
             // As an optimization, all shared statics which do not have interior
             // mutability are placed into read-only memory.
-            if !is_mutable && self.type_is_freeze(ty) {
+            if !self.tcx.is_mutable_static(def_id) && alloc.mutability.is_not() {
```

Comment:
I think just looking at `alloc` here should suffice? The interner carefully set up the mutability so that it can be directly used by codegen.